### ISSUE TITLE
The new vmWithDisko command doesn't work with impermanence enabled

### ISSUE THREAD

**Issue by arjan-s** [nix-community/disko#763]
All my systems are configured with both disko and impermanence enabled, and I'd like to use the new `.#nixosConfigurations.vm.config.system.build.vmWithDisko` ([docs](https://github.com/nix-community/disko/blob/master/docs/interactive-vm.md)) command to test config changes for them.

This combination currently causes the following error, even though the mentioned `neededForBoot` is set correctly:
```
error:
       … while calling the 'head' builtin

         at /nix/store/5w3dp0m37794iffsbm9vd9f1xmmhda6i-source/lib/attrsets.nix:1575:11:

         1574|         || pred here (elemAt values 1) (head values) then
         1575|           head values
             |           ^
         1576|         else

       … while evaluating the attribute 'optionalValue.value'

         at /nix/store/5w3dp0m37794iffsbm9vd9f1xmmhda6i-source/lib/modules.nix:868:5:

          867|
          868|     optionalValue =
             |     ^
          869|       if isDefined then { value = mergedValue; }

       (stack trace truncated; use '--show-trace' to show the full trace)

       error:
       Failed assertions:
       - environment.persistence:
           All filesystems used for persistent storage must
           have the flag neededForBoot set to true.

           Please fix or remove the following paths:
             /persist
```

As discussed on Matrix, here is a relatively small self-contained flake that reproduces the issue.

```nix
{
  inputs = {
    nixpkgs.url = "nixpkgs/nixos-unstable";
    impermanence.url = "github:nix-community/impermanence";
    disko = {
      url = "github:nix-community/disko/v1.7.0";
      inputs.nixpkgs.follows = "nixpkgs";
    };
  };
  outputs = {nixpkgs, ...} @ inputs: let
    inherit (nixpkgs) lib;
  in {
    inherit lib;
    nixosConfigurations = {
      vm = lib.nixosSystem {
        specialArgs = {inherit inputs;};
        modules = [
          inputs.disko.nixosModules.disko
          inputs.impermanence.nixosModules.impermanence
          {
            system.stateVersion = "24.11";
            nixpkgs.hostPlatform.system = "x86_64-linux";
            networking.hostName = "vm";
            disko.memSize = 2048;
            boot = {
              loader = {
                systemd-boot.enable = true;
                timeout = 2;
                efi.canTouchEfiVariables = true;
              };
              initrd = {
                availableKernelModules = ["xhci_pci" "thunderbolt" "nvme" "usb_storage" "sd_mod" "rtsx_pci_sdmmc"];
                kernelModules = ["dm-snapshot"];
              };
            };
            disko.devices.disk.nvme = {
              type = "disk";
              device = "/dev/nvme0n1";
              imageSize = "10G";
              content = {
                type = "gpt";
                partitions = {
                  ESP = {
                    start = "0";
                    end = "512MiB";
                    type = "EF00";
                    priority = 1;
                    content = {
                      type = "filesystem";
                      format = "vfat";
                      mountpoint = "/boot";
                    };
                  };
                  root = {
                    size = "100%";
                    priority = 2;
                    content = {
                      type = "btrfs";
                      subvolumes = {
                        root = {
                          mountpoint = "/";
                          mountOptions = ["noatime"];
                        };
                        nix = {
                          mountpoint = "/nix";
                          mountOptions = ["noatime"];
                        };
                        persist = {
                          mountpoint = "/persist";
                          mountOptions = ["noatime"];
                        };
                      };
                    };
                  };
                };
              };
            };
            fileSystems."/persist".neededForBoot = true;
            environment.persistence."/persist" = {
              hideMounts = true;
              directories = [
                "/var/lib/nixos"
                "/var/lib/systemd/coredump"
                "/var/log"
              ];
              files = [
                "/etc/machine-id"
              ];
            };
          }
        ];
      };
    };
  };
}
```

This config works correctly when the impermanence (`persistence`) bits are removed.

**Comment by Enzime** [comment]
Should be fixed by #766

```nix
{
  virtualisation.vmVariantWithDisko.virtualisation.fileSystems."/persist".neededForBoot = true;
}
```

**Comment by arjan-s** [comment]
Yes! It works perfectly now! Thank you!
